### PR TITLE
Add support for multiple pod networks

### PR DESCRIPTION
**Pull request by dcbw**
PodNetwork has a new member "Networks" which takes an array of
strings. These map to CNI configuration network names (the "name"
field of CNI config JSON).

If Networks is left empty, then the previous behavior of using
the first found config in /etc/cni/net.d still happens.

Signed-off-by: Dan Williams <dcbw@redhat.com>

@mrunalp @rajatchopra what do you guys think?

### CODE REVIEW

File: pkg/ocicni/ocicni.go [hunk 1]
```diff
@@ -222,27 +236,39 @@ func getDefaultCNINetwork(pluginDir string, cniDirs []string, vendorCNIDirPrefix
...
-		logrus.Infof("CNI network %s (type=%v) is used from %s", confList.Name, confList.Plugins[0].Network.Type, confFile)
+		if confList.Name == "" {
+			confList.Name = path.Base(confFile)
+		}
+
+		logrus.Infof("Found CNI network %s (type=%v) at %s", confList.Name, confList.Plugins[0].Network.Type, confFile)
+
 		// Search for vendor-specific plugins as well as default plugins in the CNI codebase.
-		vendorDir := vendorCNIDir(vendorCNIDirPrefix, confList.Plugins[0].Network.Type)
+		vendorDir := vendorCNIDir(plugin.vendorCNIDirPrefix, confList.Plugins[0].Network.Type)
```
Comment (rajatchopra) [inline pkg/ocicni/ocicni.go]: Could we flatten Plugins and all the Plugin.Network.Type in the Path? Plugins[0] will still leave it selective.

**Comment by dcbw** [comment]
> Did we want to expose any other functions meanwhile? Like GetNetwork? so that the library can be used for other purposes?

Maybe let's see if anyone asks for it before we do?

File: pkg/ocicni/ocicni.go (hunk 1 above)
Comment (dcbw) [reply pkg/ocicni/ocicni.go]: > Could we flatten Plugins and all the Plugin.Network.Type in the Path? Plugins[0] will still leave it selective.

@rajatchopra done

**Comment by dcbw** [comment]
Holding off on proposing this for merge until CNI https://github.com/containernetworking/cni/pull/558 lands.

**Comment by dcbw** [comment]
@baude @mheon @rajatchopra @cri-o/ocicni-maintainers PTAL thanks!

**Comment by rhatdan** [comment]
Thought this was under CRI-O...

**Comment by mrunalp** [comment]
@dcbw Can your rebase?

**Comment by dcbw** [comment]
@mrunalp done, new commit added (missing default network...)

**Comment by rhatdan** [comment]
@dcbw Tests think you need to gofmt your code.

**Comment by dcbw** [comment]
@rhatdan @mrunalp done

**Comment by rhatdan** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Comment by runcom** [comment]
vendoring this in crio produces the following:
```
go build -i -ldflags '-s -w -X main.gitCommit="47ede695dffc2abd2e7bbf5e4f68ccb78a17fae5-dirty" -X main.buildInfo=1548764930' -tags "     seccomp selinux  containers_image_ostree_stub" -o bin/crio github.com/kubernetes-sigs/cri-o/cmd/crio
# github.com/kubernetes-sigs/cri-o/server
server/sandbox_network.go:29:14: cannot assign []"github.com/kubernetes-sigs/cri-o/vendor/github.com/containernetworking/cni/pkg/types".Result to result (type "github.com/kubernetes-sigs/cri-o/vendor/github.com/containernetworking/cni/pkg/types".Result) in multiple assignment:
	[]"github.com/kubernetes-sigs/cri-o/vendor/github.com/containernetworking/cni/pkg/types".Result does not implement "github.com/kubernetes-sigs/cri-o/vendor/github.com/containernetworking/cni/pkg/types".Result (missing GetAsVersion method)
server/sandbox_network.go:36:13: cannot assign []"github.com/kubernetes-sigs/cri-o/vendor/github.com/containernetworking/cni/pkg/types".Result to podIP (type string) in multiple assignment
server/sandbox_network.go:76:2: cannot use ip (type []"github.com/kubernetes-sigs/cri-o/vendor/github.com/containernetworking/cni/pkg/types".Result) as type string in return argument
make: *** [Makefile:105: bin/crio] Error 2
```
@dcbw could you help me out? :angel:

**Pull request merged**